### PR TITLE
[Siege] add support for omniscience and it's unique

### PR DESCRIPTION
**Pull request by learn2draw**
Adds support for omniscience

Pastebin for testing
https://pastebin.com/6kQxHN5g

image
image

### CODE REVIEW

**Comment by blvcksvn** [comment]
image

Error with Mjolner requirement override, should be displaying Omni required as 1343, not 978.

Not sure if this is addressed, but the actual Omniscience cost at 23% satisfy is 1791/1304
Omniscience should also be addressed as an "or" case as the base attributes can still fulfil attribute requirements for some low level items

**Comment by learn2draw** [comment]
> Error with Mjolner requirement override, should be displaying Omni required as 1343, not 978.

Ahh, it was overwriting the higher STR requirement with the lower INT one in the breakdown, fixed it to stop from overwriting with lower values.
 
> Not sure if this is addressed, but the actual Omniscience cost at 23% satisfy is 1791/1304. Formula is AR * (100/Satisfy%) Omniscience should also be addressed as an "or" case as the base attributes can still fulfil attribute requirements for some low level items

I had `Items and Gems have 25% reduced Attribute Requirements` to test it with other reduced attribute stuff which would be 
`(412 * 0.75) * (100/23) = 1343`
If I remove this line it results in 1791 requirement

File: src/Classes/ItemsTab.lua [hunk 1]
```diff
@@ -2563,8 +2563,8 @@ function ItemsTabClass:AddItemTooltip(tooltip, item, slot, dbMode)
 
 	-- Requirements
 	self.build:AddRequirementsToTooltip(tooltip, item.requirements.level, 
-		item.requirements.strMod, item.requirements.dexMod, item.requirements.intMod, 
-		item.requirements.str or 0, item.requirements.dex or 0, item.requirements.int or 0)
+		item.requirements.strMod, item.requirements.dexMod, item.requirements.intMod, item.requirements.omniMod, 
+		item.requirements.str or 0, item.requirements.dex or 0, item.requirements.int or 0, item.requirements.omni or 0)
```
Comment (Wires77) [inline src/Classes/ItemsTab.lua:2567]: I don't think this is supposed to be here, or you forgot to add these params to the method signature?

Comment (learn2draw) [reply src/Classes/ItemsTab.lua]: Addressed this to properly do the tooltip requirement for Omniscience 
image

**Review by Wires77 — APPROVED** [review]
LGTM

**Pull request merged**